src/Utilities/cachingoptimizer.jl

```diff
@@ -168,7 +168,7 @@ function MOI.addvariable!(m::CachingOptimizer)
             try
                 vindex_optimizer = MOI.addvariable!(m.optimizer)
             catch err
-                if err isa MOI.CannotAddVariable
+                if err isa MOI.NotAllowedError
```

Comment:
What's the motivation for not checking for the specific error type like `AddVariableNotAllowed`? If the wrong error type like `DeleteNotAllowed` is thrown, that's an error that shouldn't silently be ignored.

Reply:
That may allow the user to create new error types. Also, trying to do one modification might result in another one and hence another error.
For instance, if you want to add a constraint and the bridge try to add a variable then it will throw a `AddVariableNotAllowed`.
We have already discussed this and I know you don't like that but if we start doing try catch everywhere, we loose the advantage of having removed `canget`. The advantage of errors is that you don't have to check that it is allowed and you just do the operation. (Except the cachingoptimizer)

Reply:
I'm not a big fan but ok.